starknet_py/utils/merkle_tree.py

```diff
@@ -0,0 +1,52 @@
+from dataclasses import dataclass, field
+from typing import List, Tuple
+
+from starknet_py.hash.hash_method import HashMethod
+from starknet_py.net.client_utils import _to_rpc_felt
+
+
+@dataclass
+class MerkleTree:
+    """
+    Dataclass representing a MerkleTree object.
+    """
+
+    leaves: List[str]
+    hash_method: HashMethod
+    root_hash: str = field(init=False)
+    branches: List[List[str]] = field(init=False)
+
+    def __post_init__(self):
+        if self.leaves:
+            self.root_hash, self.branches = MerkleTree.build(
+                self.leaves, self.hash_method
+            )
+
+    @staticmethod
+    def build(
+        leaf_hashes: List[str], hash_method: HashMethod
+    ) -> Tuple[str, List[List[str]]]:
+        if not leaf_hashes:
+            raise ValueError("Cannot build Merkle tree from an empty list of leaves.")
+
+        leaves = leaf_hashes[:]
+        branches: List[List[str]] = []
+
+        while len(leaves) > 1:
+            if len(leaves) != len(leaf_hashes):
+                branches.append(leaves[:])
+
+            new_leaves = []
+            for i in range(0, len(leaves), 2):
+                a, b = leaves[i], leaves[i + 1] if i + 1 < len(leaves) else "0"
+                a, b = int(a, 16), int(b, 16)
+                new_leaves.append(_to_rpc_felt(MerkleTree.hash(a, b, hash_method)))
```

Comment:
Do we need the static function `MerkleTree.hash` at all?
```suggestion
                new_nodes.append(_to_rpc_felt(hash_method.hash(*sorted([a, b]))))
```

Reply:
1. I think it's better to keep sorting `a` and `b` inside the method, rather than passing result of `sorted()` as an arg.  Using `hash_method.hash()` assumed that args will be passed on proper order.
2. `hash_method` has `hash()` and `hash_many()` which may pose some uncertainty while using.
3. Other SDKs have same method, so it would be good to keep the consistency.

Wdyt @ddoktorski ?

Reply:
The main question is: do we really need the static `hash` function in the `MerkleTree`? Is it going to be useful for users?

Reply:
Offline discussion: we decided to remove `MerkleTree.hash()` as it didn't simplify much and shouldn't be used by users.